tools/pre-commit-hooks/check_version_consistency.py

```
@@ -0,0 +1,163 @@
+    except (OSError, ValueError) as e:
+        print(f"ERROR: Failed to read version files: {e}", file=sys.stderr)
+        return False
+
+    # Regenerate version.h from VERSION.md (the source of truth).
+    # pre-commit detects the file modification and asks the developer to stage it.
+    _regenerate_version_header(base_path, version_md)
+
+    if verbose:
+        print(f"VERSION.md:             {version_md}")
+        print(f"config.py:              {config_py_version}")
+        print(f"version.h:              regenerated from VERSION.md")
```

Ruff (0.15.9)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)